shiny/render/renderer/_renderer.py

```diff
@@ -203,23 +255,39 @@ def __init__(
 
     async def transform(self, value: IT) -> Jsonifiable:
         """
-        Renderer - transform docs here
-
-        TODO-barret-docs
+        Transform an non-`None` output value into a JSON-serializable object.
+
+        If an `.render()` method is not implemented, this method **must** be
+        implemented. When requested, the `.render()` method's default behavior will
+        resolve the output value function, return `None` if the value is `None`, and
+        call this method to transform the value into a JSON-serializable object. Note,
+        only one of `.transform()` or `.render()` should be implemented. If both are
+        implemented, `.render()` will be used.
         """
         raise NotImplementedError(
-            "Please implement either the `transform(self, value: IT)` or `render(self)` method.\n"
-            "* `transform(self, value: IT)` should transform the `value` (of type `IT`) into Jsonifiable object. Ex: `dict`, `None`, `str`. (standard)\n"
-            "* `render(self)` method has full control of how an App author's value is retrieved (`self._fn()`) and utilized. (rare)\n"
-            "By default, the `render` retrieves the value and then calls `transform` method on non-`None` values."
+            "Please implement either the `transform(self, value: IT)`"
+            " or `render(self)` method.\n"
+            "* `transform(self, value: IT)` should transform the `value` (of type `IT`)"
+            " into Jsonifiable object. Ex: `dict`, `None`, `str`. (common)\n"
+            "* `render(self)` method has full control of how an App author's value is"
+            " retrieved (`self._fn()`) and processed. (rare)\n"
+            "By default, the `render` retrieves the value and then calls `transform`"
+            " method on non-`None` values."
         )
 
     async def render(self) -> Jsonifiable:
         """
-        Renderer - render docs here
+        Renders the output value function.
+
+        This method is called when the renderer is requested to render its output. It
+        should resolve the value function `.fn` and return a (most likely) transformed
```

Comment:
I'm a little confused why you're using `resolve` instead of `call`/`execute`?

Reply:
I'm up for `call`!

```suggestion
        should call the value function `.fn` and return a (most likely) transformed
```

Reply:
Switching to `execute` 😆 
```suggestion
        should execute the value function `.fn` and return a (most likely) transformed
```

Reply:
Cool, I think it gets used in a few other places too